src/Symfony/Component/DependencyInjection/Compiler/ResolveAutowireInlineAttributesPass.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Attribute\AutowireInline;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\VarExporter\ProxyHelper;
+
+/**
+ * Inspects existing autowired services for {@see AutowireInline} attribute and registers the definitions for reuse.
+ *
+ * @author Ismail Özgün Turan <oezguen.turan@dadadev.com>
+ */
+class ResolveAutowireInlineAttributesPass extends AbstractRecursivePass
+{
+    protected bool $skipScalars = true;
+
+    protected function processValue(mixed $value, bool $isRoot = false): mixed
+    {
+        $value = parent::processValue($value, $isRoot);
+
+        if (!$value instanceof Definition || !$value->isAutowired() || $value->isAbstract() || !$value->getClass()) {
+            return $value;
+        }
+
+        if (!$reflectionClass = $this->container->getReflectionClass($value->getClass(), false)) {
+            $this->container->log($this, sprintf('Skipping service "%s": Class or interface "%s" cannot be loaded.', $this->currentId, $value->getClass()));
+
+            return $value;
+        }
+
+        $constructorReflectionMethod = $reflectionClass->getConstructor();
```

Comment:
this should call `$this->getConstructor($value, false)` instead, and we should also parse `$value->getMethodCalls()`
Basically we should visit the same parameters as we do in AutowirePass.

Reply:
Done [here](https://github.com/symfony/symfony/compare/dd16390a5290a9a08b2c997fa9d959296bb6f0e0..c94322321acafdbaf32044e58b213223645497ce) but still not sure if this was now done correctly by me. Do I have to process the parameters there as well similar to what `AutowirePass` does internally with the `methodCalls` and so on?